README.md

```diff
@@ -1 +1,230 @@
-Magento_UpwardConnector module used for correctly routing frontend requests through UPWARD-PHP.
+# How to install UPWARD with a PWA Studio storefront
+
+## Prerequisites
+
+- An existing Magento Cloud project
+- Composer
+- Yarn or NPM
+- A [Magento PWA Studio][] storefront managed by NPM or Yarn.
+
+  You are not required to publish to npmjs.com, but
+  NPM or Yarn should be able to access your project code through Git.
+
+  For this tutorial, the [`@magento/venia-concept`][] package for the [Venia storefront][] is used, but any PWA available though NPM with an UPWARD compatible YAML file is supported.
+
+## Add node package dependency
+
+If it does not exist, [create a `package.json`][] file in your project directory.
+
+```sh
+npm init -y
+```
+
+Add your PWA Studio storefront to the list of dependencies in this file.
+
+```json
+{
+  "dependencies": {
+    "@magento/venia-concept": "~2.0.0"
+  }
+}
+```
+
+## Add Composer dependencies
+
+### Option 1: Git based
+
+Add the repository information for the UPWARD PHP server and UPWARD Connector to the `repositories` section in your `composer.json`.
+
+```json
+"upward-connector": {
+    "type": "vcs",
+    "url": "https://github.com/magento-research/magento2-upward-connector"
+},
+"upward-php": {
+    "type": "vcs",
+    "url": "https://github.com/magento-research/upward-php"
+}
+```
+
+To have composer install these packages, add them to the `require` section of the `composer.json` file.
+
+```json
+"magento/module-upward-connector": "^1.0.0",
+"magento/upward": "^1.0.0"
+```
+
+### Option 2: Package based
+
+Add the repository information for the Magento package service to the `repositories` section in your `composer.json`.
+
+```json
+"repo": {
+    "type": "composer",
+    "url": "https://repo.magento.com/"
+}
+```
+
+Add package to the `require` section of the `composer.json` file.
+
+```json
+"magento/module-upward-connector": "^1.0.0"
+```
```

Comment:
I would say we treat `README` as a living document. As of right now, doing a Git based deployment is the only option so it should be the documented process.

In the future, following a package based process is best practice so it should be the documented flow; git based may be used for developers who want to use pre-release versions and that process is documented in other places.